### PR TITLE
Simplified code

### PR DESCRIPTION
**Pull request by Devit951**
Here changes:
- Named function argument instead of creating a new one variable.
- `requireContext()` instead of manually check the context to null
- `context.dimension(R.dimen.image_size).toInt()` extension instead of `context.resources.getDimension(R.dimen.image_size).toInt()`

### CODE REVIEW

File: library_base/src/main/java/com/igorwojda/showcase/base/presentation/extension/ContextExtensions.kt [hunk 1]
```diff
@@ -20,3 +22,5 @@ inline fun <reified T : AppCompatActivity> Context.startActivity(
     intent.putExtras(bundleOf(*extras))
     startActivity(intent)
 }
+
+fun Context.dimension(@DimenRes dimenRes: Int) = resources.getDimension(dimenRes)
```
Comment (igorwojda) [inline library_base/src/main/java/com/igorwojda/showcase/base/presentation/extension/ContextExtensions.kt]: I don't think this is the correct name. 

This is function that defines an action and I prefer to follow `doSomething` convention (eg. `clearCache`, `getUser`, `setData`, `removeItem`...). My preference comes mainly from the fact that in Kotlin (as opposed to Java) we also have properties that describe the sate (`name`, `size`, `dimension`).

Please take a look at this
https://blog.kotlin-academy.com/kotlin-should-i-define-function-or-property-6786951da909

File: feature_album/src/main/java/com/igorwojda/showcase/feature/album/presentation/albumlist/AlbumListFragment.kt [hunk 2]
```diff
@@ -23,24 +24,20 @@ class AlbumListFragment : BaseContainerFragment() {
...
-        }
-
         recyclerView.apply {
             setHasFixedSize(true)
-            val columnWidth = context.resources.getDimension(R.dimen.image_size).toInt()
             layoutManager =
                 GridAutofitLayoutManager(
-                    context,
-                    columnWidth
+                    requireContext(),
```
Comment (igorwojda) [inline feature_album/src/main/java/com/igorwojda/showcase/feature/album/presentation/albumlist/AlbumListFragment.kt]: It's a bit hard to spot it, but there is a bit of inconsistency on Kotlin vs Android.

In Kotlin you have two kinds of methods:
`requite()`/`requireNotNull()` that may throw in `IllegalArgumentException`
`check()`/`checkNotNull()` that may throw in `IllegalStateException`

vs in Android you have
`requireContext()` that may throw in `IllegalStateException` (ideally it should be named `checkContext`)

It's quite common to verify arguments and state across project, so I think that distinction is nice and thus I prefer to follow Kotlin approach across the project (to keep it consistent).

I guess I we could remove this local `context` variable, but I would prefer to use `checkNotNull(context)`

**Comment by igorwojda** [comment]
Thx for the PR
I have written a detailed explanation and I am open for discussion, but IMO this proposal does not really fit into the project.

BTW When you create PR it's nice to make sure that all CI checks are passing.

**Comment by Devit951** [comment]
Thank you for the detailed response! I think, like you, that this PR doesn't fit into this project.

**Pull request closed without merging**